### ISSUE TITLE
Multiple day events show as inactive

### ISSUE THREAD

**Issue by ffes** [plusonelabs/calendar-widget#184]
I have a single multi day event, let's say from Monday till Friday, in my calendar. On Tuesday it is shown as a inactive event on Monday **and** as an active event on Tuesday. On Wednesday it still shows as an inactive event on Monday and as active on Wednesday, etc, etc. For event longer that a couple of days, like a period of vacation for our scholar son, it will show up as inactive on the start date for over a month.

**Comment by dcanl** [comment]
Same problem here! 

The bug appeared since last update (installed 30 minutes ago), which introduced the "show events that ended recently" feature. 

For now, best thing I found is to tick this new functionality, and to untick the option which adds an entry on every day for a multiple day event. Not 100% satisfying, but better than double entries. 

Please, correct the bug! :-) 

And thanks for your work, I love your widget!

**Comment by mpost** [comment]
We are aware of the issue and working on a fix for the upcoming 1.9.1 release. Please stay tuned.

**Comment by yvolk** [comment]
I think that we should start from creating a test case for this issue  (and it will be easier to do the same for other non - trivial cases ).
The test will allow us to understand / to agree, what actually a correct behavior should be...

**Comment by yvolk** [comment]
So, as I understand,  
1. If "Fill all day events" option is turned off and we are inside the period, we should see this event as active and only today.

**Comment by dcanl** [comment]
For me, if "fill all day events" is turned on, we shouldn't see any disabled event at the top of the list while we're inside the period. Just 1 identical active event on each day concerned. 
If turned off, current behavior seems correct (1 event at the top of the list which tells the period of the event, active if starting today, inactive if in the period). 
    


-------- Message d'origine --------
De : Yuri Volkov <notifications@github.com> 
Date : 29/08/2015  19:08  (GMT+01:00) 
À : plusonelabs/calendar-widget <calendar-widget@noreply.github.com> 
Cc : dcanl <dcanl21-public@yahoo.fr> 
Objet : Re: [calendar-widget] Multiple day events show as inactive (#184) 

So, as I understand,


1. If "Fill all day events" option is turned off and we are inside the period, we should see this event as active and only today.

**Comment by mjt543** [comment]
I again +1 the idea of reverting to the previous behavior. It's more important for me to see how many days longer a multi-day event continues by having it appear as an all-day event for each day it's active as opposed to having the day the multi-day event started perpetually visible in the widget. I like the active date at the top of the widget to always be the current day, not one or more days ago.

**Comment by yvolk** [comment]
@ffes @dcanl  I created automated tests for Multiple days events (as least for some cases... [see here|https://github.com/plusonelabs/calendar-widget/blob/master/app/calendar-widget/src/androidTest/java/com/plusonelabs/calendar/MultidayEventTest.java]) and fixed reproduced bugs. Now these tests passed, but if you feel that bugs are not fixed, please provide your case to developers via the new feature in a Settings -> Feedback -> "Share events for debugging".
More on the "Share events for debugging" feature here: https://github.com/plusonelabs/calendar-widget/issues/190#issuecomment-136359880

**Comment by yvolk** [comment]
Calendar widget is published to a beta testing channel.  Please join testing at https://play.google.com/apps/testing/com.plusonelabs.calendar

**Comment by ffes** [comment]
I've tested this a couple of days now and it seems to work as expected.

But I don't like display of the continuously changing start time. That you need to do this internally makes sense, but I would always show the original start time.

**Comment by mpost** [comment]
@ffes We have spotted that as well and fixed it already: https://github.com/plusonelabs/calendar-widget/issues/199

**Comment by ffes** [comment]
@mpost Great to hear. I could/should have found that issue myself ;-)

**Comment by yvolk** [comment]
@ffes Another Beta released with a fix for #199. If you already joined the Calendar widget beta channel at https://play.google.com/apps/testing/com.plusonelabs.calendar you should now see v.1.9.2-6b6d980

**Comment by ffes** [comment]
I tested the new beta and it work fine.

One small cosmetic thing though:

In my calendar last weekend was an appointment that started Friday at 19:00 and ended Sunday at 15:00

On Saturday the end time event was shows as two arrows on Sunday. On Sunday it was as expected. Can't remember what it looked like on Friday.

I attached two screenshots. The pictures say at all.

![saturday](https://cloud.githubusercontent.com/assets/3225656/10052702/8347ae92-6228-11e5-85fd-91c3d0bc0ee5.png)
![sunday](https://cloud.githubusercontent.com/assets/3225656/10052703/834801c6-6228-11e5-9750-895c78da12a5.png)
